### PR TITLE
Re-design feature set

### LINKED ISSUE #109: Re-work exposed features
I am using this issue to create two proposals:

1. We should use the new `dep:` syntax for features to hide the implicit features created by optional crates. I think this part is non-controversial.
2. Rename the features we expose to their original crate name. The argument would be that we are currently not supporting multiple versions of the same dependency and thus the `with-xyz-1_0` convention is unnecessarily complex.

Together with this issue, the README should be updated too.

### PR DESCRIPTION
**Pull request by thomaseizinger**
Fixes #109.

This also makes use of the new `dep:` syntax and thus hides the implicit features generated by optional dependencies.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
         toolchain: [
           stable,
           beta,
-          1.56.0 # MSRV
+          1.60.0 # MSRV
```
Comment (Restioson) [inline .github/workflows/ci.yml:15]: Out of interest, what is the reason for this bump?

File: Cargo.toml [hunk 2]
```diff
@@ -54,69 +47,70 @@ async-std = { version = "1.0", features = ["attributes"] }
...
-instrumentation = ["tracing"]
-with-async_std-1 = ["async-std"]
-with-smol-1 = ["smol"]
-with-tokio-1 = ["tokio"]
-with-wasm_bindgen-0_2 = ["wasm-bindgen", "wasm-bindgen-futures"]
+instrumentation = ["dep:tracing"]
+async_std = ["dep:async-std"]
+smol = ["dep:smol"]
+tokio = ["dep:tokio"]
+wasm_bindgen = ["dep:wasm-bindgen", "dep:wasm-bindgen-futures"]
```
Comment (Restioson) [inline Cargo.toml:54]: This is nice!

File: README.md [hunk 3]
```diff
@@ -73,22 +73,22 @@ Too verbose? Check out the [spaad](https://crates.io/crates/spaad) sister-crate!
...
 ## Okay, sounds great! How do I use it?
 Check out the [docs](https://docs.rs/xtra) and the [examples](https://github.com/Restioson/xtra/blob/master/examples)
-to get started! Enabling the `with-tokio-1`, `with-async_std-1`, `with-smol-1`, or `with-wasm_bindgen-0_2` features
+to get started! Enabling the `tokio`, `async_std`, `smol`, or `wasm_bindgen` features
 is recommended in order to enable some  convenience methods (such as `Actor::spawn`). Which you enable will depend on
 which executor you want to use (check out their docs to learn more about each). If you have any questions, feel free to
 [open an issue](https://github.com/Restioson/xtra/issues/new) or message me on the [Rust discord](https://bit.ly/rust-community).
 
-Keep in mind that `xtra` has a MSRV of 1.56.0.
+Keep in mind that `xtra` has a MSRV of 1.60.0.
```
Comment (Restioson) [inline README.md:81]: This should be updated in Cargo.toml

**Review by Restioson — COMMENTED** [review]
Thanks! Other than the one thing I mentioned this is ready for merge IMO

File: .github/workflows/ci.yml (hunk 1 above)
Comment (thomaseizinger) [reply .github/workflows/ci.yml:15]: The `dep:` syntax! It was only stabilised in 1.60.0.

**Comment by thomaseizinger** [comment]
Conflicts resolved and comments addressed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Restioson) [reply .github/workflows/ci.yml:15]: Ah, I see

**Comment by Restioson** [comment]
Thanks!

**Pull request merged**